app/src/main/java/com/simplecity/amp_library/playback/MusicService.java

```diff
@@ -1753,7 +1760,17 @@ public void openFile(String path, @Nullable Action completion) {
      */
     public void play() {
 
-        int status = audioManager.requestAudioFocus(audioFocusListener, AudioManager.STREAM_MUSIC, AudioManager.AUDIOFOCUS_GAIN);
+        int status;
+        if (ShuttleUtils.hasOreo()) {
```

Comment:
Good change, should probably be made under a separate commit. It's very hard to review among all these other changes.